src/vision/status_quo/alan_finds_database_drops_hard.md

```diff
@@ -0,0 +1,84 @@
+# 😱 Status quo stories: Alan finds dropping database handles is hard.
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+
+## The problem
+
+Alan has just written a function that extends YouBuy to connect to a database when his application receives a request. The request returns all the data for a single user to a customized page on his website. After launching the beta version of his application, Alan begins receiving complaints that his website sometimes displays a page with an error message and other times it seems to work as intended. When the page with an error message appears the message reads: "Can not open a connection to the database, because a connection is already open".
+
+```rust=
+use sqlx::Connection;
+
+#[async_std::main]
+async fn main() -> Result<(), sqlx::Error> {
+    // Create a connection
+
+    let conn = SqliteConnection::connect("sqlite::memory:").await?;
+
+    // Make a simple query to return the given parameter
+    let row: (i64,) = sqlx::query_as("SELECT $1")
+        .bind(150_i64)
+        .fetch_one(&conn).await?;
+
+    assert_eq!(row.0, 150);
+
+    Ok(())
+}
+```
+
+Alan sees the reports coming in from his beta users, but hasn't received any alarms from his application. He realizes he must start debugging. His initial testing produces the customized page for the current user as expected, but after a number of attempts he sees the page with the error message. His application shows an error that says that there is already an open database handle.
+
+
+## Searching for the Solution
+
+
+Alan tries to figure out what happened from the logs, but is unable to do so. Alan decides to attempt to recreate the scenario by doing extra load testing and trying to emulate the increased traffic. After some time, Alan gets lucky and is able to recreate the error, but he still doesn't know what could be the cause. Alan turns to the documentation for the `sqlx` crate to see if there are flags that might enable extra instrumentation but he can't find any (XXX check what crate offers).
+
+Alan, next, takes the following steps to determine the most likely source to explain the problem behavior in his application:
```

Comment:
@galich in particular, we were trying to imagine how you debugged the problem -- what actually happened, if you recall?

Reply:
In our case debugging was a bit painful because the system is fairly large, with many moving parts and a lot of async code. So initially we went on a few false leads, thinking that there is a problem of how we line up futures for execution or that there is a rogue process touching database or there is a migration running, etc.

Eventually we thought that we are not closing connection properly, so we double checked that we do drop before another task is trying to access it. When all was verified we decided to look into the fact that sqlx is based on async itself and started to look into the code, eventually discovering that some drops spawn new futures to actually close connections. That was totally unexpected behaviour.

At some point in time it became really strange, if I remember correctly, we saw that there are async spawns inside drops in sqlx, but I think sqlite was not using them. That was confusing, but then we remembered that we use connection pooling which has some async spawning in drop handler https://github.com/launchbadge/sqlx/blob/0ed524d65c2a3ee2e2a6706910b85bf2bb72115f/sqlx-core/src/pool/connection.rs#L74 .

Reply:
OK. I think the story is *reasonably* close to this, at least.

Reply:
changes committed